fix: hide `node_modules` and `vendor` from update output

SisyphusZheng opened the pull request:
Added logic to filter out files in node_modules and vendor directories from user-facing logs and migration summaries in the update process. Updated tests to verify that these files are processed but not shown in logs.

close #3082

SisyphusZheng committed:
Hide node_modules and vendor from update logs

Added logic to filter out files in node_modules and vendor directories from user-facing logs and migration summaries in the update process. Updated tests to verify that these files are processed but not shown in logs.

SisyphusZheng committed:
fmt

SisyphusZheng committed:
Normalize path separators in modified files log

Replaces backslashes with forward slashes in file paths when logging modified files to ensure consistent output across platforms.

marvinhagemeister commented on update/src/update_test.ts:
```diff
@@ -365,9 +366,10 @@ Deno.test(
+Deno.test({
+  ignore: true,
+  name: "update - 1.x update handler signature variable",
+  fn: async () => {
```
This seems like an unrelated change. Can you revert it?

SisyphusZheng replied:
To be honest, this issue was a format error I encountered when I cloned and ran deno test, but it seems to be fine now. Sorry for the trouble.

marvinhagemeister commented on update/src/update_test.ts:
```diff
@@ -398,11 +400,12 @@ Deno.test.ignore(
+Deno.test({
+  ignore: true,
+  name: "update - 1.x update handler signature non-inferred",
+  fn: async () => {
```
This seems like an unrelated change. Can you revert it?

marvinhagemeister commented on update/src/update_test.ts:
```diff
@@ -426,11 +429,12 @@ export const handler = {
+Deno.test({
+  ignore: true,
+  name: "update - 1.x update handler signature with destructure",
+  fn: async () => {
```
Unrelated, please revert

marvinhagemeister commented on update/src/update_test.ts:
```diff
@@ -557,9 +561,10 @@ export default async function Index(ctx: FreshContext) {
+Deno.test({
+  ignore: true,
+  name: "update - 1.x ctx.renderNotFound() -> ctx.throw()",
+  fn: async () => {
```
unrelated

marvinhagemeister commented on update/src/update_test.ts:
```diff
@@ -593,11 +598,12 @@ export const handler: Handlers = {
+Deno.test({
+  ignore: true,
+  name: "update - 1.x ctx.remoteAddr -> ctx.info.remoteAddr",
+  fn: async () => {
```
unrelated

marvinhagemeister commented on update/src/update_test.ts:
```diff
@@ -627,14 +633,17 @@ export const handler: Handlers = {
+    const dir = _tmp.dir;
+    await writeFiles(dir, {
+      "/deno.json": `{}`,
+      "/routes/index.tsx": `import { Handlers } from "$fresh/server.ts";
```
unrelated

marvinhagemeister commented on update/src/update.ts:
```diff
@@ -10,6 +10,13 @@ export const FRESH_VERSION = "2.0.0-alpha.37";
+  return normalizedPath.includes("/node_modules/") ||
+    normalizedPath.includes("/vendor/");
+}
+
```
This can be simplified to:

```suggestion
const HIDE_FILES = /[\\/]+(node_modules|vendor)[\\/]+/;
```
And then later do `!HIDE_FILES.test(someFilePath)`.

SisyphusZheng replied:
cool, thx

SisyphusZheng committed:
Update update/src/update.ts

Co-authored-by: Marvin Hagemeister <marvinhagemeister50@gmail.com>

SisyphusZheng committed:
Refactor file filtering and update test usage

Replaces shouldHideFromLogs with HIDE_FILES RegExp for file filtering in update.ts. Refactors Deno.test usage in update_test.ts to use Deno.test.ignore for ignored tests and simplifies test definitions.

marvinhagemeister reviewed (approved):
Sweet, thanks!
